node/src/Unison/Runtime/ExtraBuiltins.hs

```diff
@@ -37,17 +41,35 @@ makeRandomHash genVar = do
   _ <- MVar.swapMVar genVar newGen
   pure hash
 
+makeRandomBS :: RandomGen r => MVar.MVar r -> IO ByteString
+makeRandomBS genVar = Hash.toBytes <$> makeRandomHash genVar
+
+makeRandomSeries :: RandomGen r => MVar.MVar r -> IO Series
+makeRandomSeries genVar = Series <$> makeRandomBS genVar
+
+makeRandomAddress :: RandomGen r => MVar.MVar r -> IO Address.Address
+makeRandomAddress genVar = Address.Address <$> makeRandomBS genVar
+
+makeRandomIdentifier :: RandomGen r => MVar.MVar r -> IO (Series, Series)
+makeRandomIdentifier genVar = do
+  cp <- makeRandomSeries genVar
+  ud <- makeRandomSeries genVar
+  pure (cp, ud)
+
+
 makeAPI :: IO (WHNFEval -> [Builtin])
```

Comment:
A couple things -

Modify this function to take a `Cryptography` and a `BlockStore` and update callers of this function accordingly. The Unison libraries should always have these dependencies injected, they should never assume any particular implementations. You should be polymorphic in the `BlockStore address` type as well - do not assume the concrete `Address` type you added; instead add typeclass constraints or pass in functions if you need the ability to manipulate the `address` values of a `BlockStore address`.

Once you've threaded this through, switch to using the `Cryptography` for generating secure randomness (the `randomBytes` function), and also use it for encryption of all blocks. Hook up encryption now - we can call it with the noop `Cryptography` for testing.